plugin.video.srgssr_ch_replay/resources/lib/plugin.py

```diff
@@ -0,0 +1,542 @@
+import sys
+import os
+from string import ascii_lowercase
+from collections import namedtuple
+from typing import Tuple
+from urllib.parse import parse_qsl, quote_plus, urlparse
+import requests
+
+import xbmc
+import xbmcaddon
+import xbmcgui
+import xbmcplugin
+import xbmcvfs
+import inputstreamhelper
+
+from resources.lib.utils import to_bool
+from resources.lib.settings import Settings
+from resources.lib.router import Router
+from resources.lib.logger import Logger
+from resources.lib.srgssr_api_client import (
+    SRGSSRVideoApiClient,
+    SRGSSRSubtitlesApiClient,
+    InvalidCredentialsException,
+)
+
+
+MainMenuItem = namedtuple("MainMenuItem", ["name", "url", "icon"])
+
+
+class Plugin:
+    """Plugin's main class"""
+
+    KODI_VERSION_MAJOR = int(xbmc.getInfoLabel("System.BuildVersion").split(".", maxsplit=1)[0])
+    ADDON = xbmcaddon.Addon()
+    ADDON_ID = ADDON.getAddonInfo("id")
+    ADDON_URL = f"plugin://{ADDON_ID}"
+    HANDLE = int(sys.argv[1])
+    SRG_API_BASE_URL = "https://api.srgssr.ch"
+
+    settings = Settings()
+
+    def __init__(self):
+        self.tr = self.ADDON.getLocalizedString
+        self.icon = self.ADDON.getAddonInfo("icon")
+        self.path = self.ADDON.getAddonInfo("path")
+        self.logger = Logger(self)
+        self.router = Router(self)
+        xbmcplugin.setContent(self.HANDLE, "tv_shows")
+        self._create_work_folder()
+
+        # check if default BU is set in the settings
+        self.bu = self.settings.default_bu if self.settings.default_bu != "choose" else ""
+
+        try:
+            self.video_client, self.subs_client = self._create_api_clients()
+        except InvalidCredentialsException as exc:
+            xbmcgui.Dialog().ok(self.tr(30096).format(exc.api_name), self.tr(30097))
+            sys.exit(1)
+
+    def _create_work_folder(self):
+        """Creating the addon work folder if it doesn't exist"""
+        try:
+            userdata_path = xbmcvfs.translatePath(self.ADDON.getAddonInfo("profile")).decode(
```

Comment:
Hi @romanvm,
Thank you for the quick review and the comment. As I understand it it will be enough just to have the function:
def _create_work_folder(self):
        """Creating the addon work folder if it doesn't exist"""
        userdata_path = xbmcvfs.translatePath(self.ADDON.getAddonInfo("profile"))
        if not os.path.isdir(userdata_path):
            os.mkdir(userdata_path)

Should I quickly deliver it? Would that be ok with you?

Reply:
I tested it and delivered it. I hope it is fine.